### PR TITLE
chore: replace AttachIp() and DetachIP() by UpdateIP()

### PR DESCRIPTION
**Pull request by loicbourgois**
Internal ticket: APIGW-1716

### CODE REVIEW

File: api/instance/v1/instance_utils.go [hunk 1]
```diff
@@ -83,15 +95,35 @@ type UpdateIPRequest struct {
...
 		Reverse: req.Reverse,
-	})
+	}
+
+	switch {
+	case req.Server == nil:
+		// Do nothing
+
+	case *req.Server == "":
+		// Setup server field to detach old server
```
Comment (QuentinBrosse) [inline api/instance/v1/instance_utils.go]: Are we sure we want to use an empty string to detach a server? It was one of the reasons for which we have created `DetachIP`.
If we want to keep this behavior (`""` == detach) we should document it in the method godoc.

Comment (jerome-quere) [reply api/instance/v1/instance_utils.go]: We use the same pattern for reverse already.

Comment (QuentinBrosse) [reply api/instance/v1/instance_utils.go]: Reverse use a `*NullableStringValue`. We should use the same implementation for `Reverse` and `Server` and document it in the method godoc.

Comment (loicbourgois) [reply api/instance/v1/instance_utils.go]: If we use the same pattern, why do we have
```
	Reverse *NullableStringValue `json:"reverse"`
	Server  *string              `json:"server"`
```
Do we need to change Reverse ?
Do we need to change all `*NullableStringValue` (cf internal ticket APIGW-1721) ?

**Pull request merged**